### ISSUE TITLE
Instrument section base color

### ISSUE THREAD

**Issue by IvanMaldonado** [LMMS/lmms#2329]
The Instrument section is not the same base color as the other sections, is there any reason for this?

![qjiorzp](https://cloud.githubusercontent.com/assets/12549082/9721960/3c581e88-556a-11e5-8051-c004aa80357d.png)     
![xezj4ss](https://cloud.githubusercontent.com/assets/12549082/9721965/4ad11082-556a-11e5-8dac-23f203a1f692.png)

**Comment by Wallacoloo** [comment]
Oh, interesting. That's probably a result of Qt having a different style for whatever component we use for the file list differently than for the QVBoxLayout we probably use for the instruments.

Another inconsistency is the padding - you'll notice the instrument list is inlaid much further than the file browsers, which actually wastes a significant amount of space for something that's already this narrow.

**Comment by IvanMaldonado** [comment]
Can this be fixed? plzplzplz

**Comment by tresf** [comment]
@michaelgregorius seems to be spearheading a lot of this theming fixing, I'll ping him explicitly for you.

:+1: 

-Tres

**Comment by tresf** [comment]
I think we should also remove the text above the instruments. Drag+drop isn't spelled out for any other areas of the software and it takes up the space that two instrument plugins could occupy.

**Comment by tresf** [comment]
> Can this be fixed? plzplzplz

Upon further investigation, this may be hard-coded... https://github.com/LMMS/lmms/blob/master/src/gui/PluginBrowser.cpp#L140

**Comment by michaelgregorius** [comment]
@tresf Judging from the screenshots it looks like @IvanMaldonado does not use the default theme (check the icons for example). If that's the case I think that's something that rather needs to be fixed by the original theme author.

I think the code line that you have referenced is responsible for changing the background color of the instrument descriptions in the instrument plugins view when the mouse hovers over them. Is this something that should be changed as well?

Ripping out the text above the instruments should be a quick job although I wonder if it is really that intuitive that the instruments can be dragged and dropped onto multiple targets. Perhaps we should have something like a "What's this?" icon that does not take as much space and that can be clicked for further explanations.

**Comment by tresf** [comment]
> I think the code line that you have referenced is responsible for changing the background color of the instrument descriptions in the instrument plugins view when the mouse hovers over them. Is this something that should be changed as well?

Correct, it has hard-coded background color, which the OP is asking to change. :smile: 

> Ripping out the text above the instruments [...] a "What's this?" icon

:+1: 

> I think that's something that rather needs to be fixed by the original theme author.

Correct me if I'm wrong, but @IvanMaldonado has presumably already tried adjusting this with the theme to no avail.

**Comment by IvanMaldonado** [comment]
@tresf Is right, I tried many things but I only know how to work with the css and png files in the themes folder.

I just noticed the inconsistency and brought it here, to check if it can be changed for further versions :)

**Comment by tresf** [comment]
Here's a quick go at it via [`10233d0`](https://github.com/LMMS/lmms/compare/master...tresf:2329-instruments).

Note, this is to fulfill the OP's request for consistency, not necessarily to make it look better.

All I did was remove some hard coded colors and fonts.  If we want to keep the button styling, there will need to be more work done to expose the proper styles to the `PluginBrowser` class (something I'd much rather defer to someone like @michaelgregorius with more experience in this regard).

![screen shot 2015-09-09 at 2 25 27 am](https://cloud.githubusercontent.com/assets/6345473/9754741/1c3adf88-569a-11e5-8249-fb511c022a47.png)

versus the file browser:

![screen shot 2015-09-09 at 2 25 49 am](https://cloud.githubusercontent.com/assets/6345473/9754742/1f561dfe-569a-11e5-8976-071f429c67fa.png)

**Comment by michaelgregorius** [comment]
@tresf Here is a patch that more or less reinstates the old look but now via stylesheet (except the margins):
```
diff --git a/data/themes/default/style.css b/data/themes/default/style.css
index 1ffd0c6..2030d62 100644
--- a/data/themes/default/style.css
+++ b/data/themes/default/style.css
@@ -11,12 +11,23 @@ QMdiArea {
 	background-image: url(resources:background_artwork.png);
 }
 
+FileBrowserTreeWidget {
+	/* Lists of samples, presets, etc */
+}
+
 PluginDescList {
 	background-color: #808080;
 }
 
 PluginDescWidget {
-	color: #e0e0e0;
+	background-color: #c0c0c0;
+	color: #404040;
+	border: 1px solid rgba(0, 0, 0, 128);
+	border-radius: 4px;
+}
+
+PluginDescWidget:hover {
+	background-color: #e0e0e0;
 }
 
 AutomationEditor {
diff --git a/src/gui/PluginBrowser.cpp b/src/gui/PluginBrowser.cpp
index 7743487..9ae2534 100644
--- a/src/gui/PluginBrowser.cpp
+++ b/src/gui/PluginBrowser.cpp
@@ -31,6 +31,7 @@
 #include <QCursor>
 #include <QMouseEvent>
 #include <QScrollArea>
+#include <QStyleOption>
 
 #include "embed.h"
 #include "debug.h"
@@ -130,9 +131,16 @@ PluginDescWidget::~PluginDescWidget()
 
 
 
-void PluginDescWidget::paintEvent( QPaintEvent * )
+void PluginDescWidget::paintEvent( QPaintEvent * e )
 {
 	QPainter p( this );
+
+	// Paint everything according to the style sheet
+	QStyleOption o;
+	o.initFrom(this);
+	style()->drawPrimitive( QStyle::PE_Widget, &o, &p, this );
+
+	// Draw the rest
 	const int s = 16 + ( 32 * ( tLimit( height(), 24, 60 ) - 24 ) ) /
 								( 60 - 24 );
 	const QSize logo_size( s, s );
```
This style sheet also exposes the `FileBrowserTreeWidget` which might be of interest for theme authors.

When applied the instrument browser looks like this (I have added some radius, feel free to remove it):
![styled](https://cloud.githubusercontent.com/assets/9293269/9768560/bbec4b54-5724-11e5-98cb-987ffe79e93b.png)

**Comment by tresf** [comment]
@michaelgregorius Thanks.  Is there a reason the instrument description text is so large in your screenshot?

**Comment by michaelgregorius** [comment]
@tresf I guess it is caused by my rather high DPI display (117.5 DPI). The logo graphics stay the same size in pixels but the font is scaled according to the point size of the font. So a higher DPI display will use more pixels to display a font with a given size than a lower DPI display.

If you want to fix the size of the font for the instrument browser via the CSS please specify it in points, e.g. 9pt, instead of pixels. Otherwise it will not scale on high DPI displays and might become unreadable. Thanks!

**Comment by tresf** [comment]
> If you want to fix the size of the font for the instrument browser via the CSS please specify it in points, e.g. 9pt, instead of pixels. Otherwise it will not scale on high DPI displays and might become unreadable. Thanks!

Sure, but isn't that hard-coded currently?  Also, shouldn't we start using [`em`](http://stackoverflow.com/a/609561/3196753) instead of `pt` anyway?

-Tres

**Comment by michaelgregorius** [comment]
@tresf The only font related action in `PluginDescWidget::paintEvent` is setting the font to bold and back to normal but nothing related to the size.

I had to read up on the question regarding `em` vs. `pt`. It seems like `em` is always relative to the set font whereas the `pt` will appear the same "physical" size on different displays with different DPI. Example: The user has set `12pt` as his default font, then `1em = 12pt`, `1.5em = 18pt`, etc. So it seems that the question boils down to whether we want relative and scalable designs (the user can choose his base font and everything scales according to that) or rather "fixed" designs that are rendered the same on different displays.

**Comment by Spekular** [comment]
My vote goes to em, fwiw.

**Comment by tresf** [comment]
> @tresf The only font related action in PluginDescWidget::paintEvent is setting the font to bold and back to normal but nothing related to the size.

Ah yes, I was thinking about life before https://github.com/LMMS/lmms/commit/76e4859ef085500d961fe8af7cc6e082813e06d1 but if that is the case, mine should be larger too, no?

**Comment by michaelgregorius** [comment]
@tresf What do you mean with larger? I guess the font is inherited from one of the parents or it is the standard application font.

**Comment by tresf** [comment]
Well, if you look at my screenshot in https://github.com/LMMS/lmms/issues/2329#issuecomment-138801911, that is the desired size moving forward.  I had to fastforward your font stuff though I had an old build but if we do this change, we likely want to keep the smaller font there.

**Comment by michaelgregorius** [comment]
@tresf Currently the plugin description is one single custom widget and the CSS will only enable us to set the font size for the whole widget. One solution might be to paint the header font of the widget and then to decrease the size of the font when paining the description. However, in my opinion it would be more flexible if the widget was just composed of three labels with a layout (one label for the pixmap, one for the header, one for the text). Then it should be possible to style sheet each individual label with regards to font size, etc. Although I am not 100% sure about that.

**Comment by IvanMaldonado** [comment]
Will this be merged?

**Comment by tresf** [comment]
@IvanMaldonado there's no open pull request on this, just conversation currently.  Please feel free to use the code suggestions to create your own pull request.

**Comment by RebeccaDeField** [comment]
Any status on this? I'm glad to see that this was already being worked on, as I needed the hard coding to be removed as well.

**Comment by tresf** [comment]
It's currently stalled due to lack of developer resources at the moment. The proposals above can take care of it, but there are minor cosmetic tradeoffs in doing so that should be evaluated.

It is a great bug for someone just familiarizing with the codebase, but we are very short on help this season.

**Comment by RebeccaDeField** [comment]
@tresf I see where you are coming from. Although I am a designer, I am expanding my horizons to include a lot more such as coding. I have learned a lot recently, but not enough to remove the hard coding (and the other various tasks that I have requested help with).

I can however, help to come up with a design for the current Instrument Section based on what @michaelgregorius posted that matches the look of the current default theme. Let me know if that would be helpful :)
